platform/default/src/mbgl/gfx/headless_frontend.cpp

```diff
@@ -140,28 +140,26 @@ PremultipliedImage HeadlessFrontend::readStillImage() {
     return backend->readStillImage();
 }
 
-HeadlessFrontend::RenderResult HeadlessFrontend::render(Map& map) {
+HeadlessFrontend::RenderResult HeadlessFrontend::render(Map& map,
+                                                        std::exception_ptr *error) {
     HeadlessFrontend::RenderResult result;
-    std::exception_ptr error;
     gfx::BackendScope guard { *getBackend() };
 
     map.renderStill([&](const std::exception_ptr& e) {
```

Comment:
Is the callback passed to `renderStill` called multiple times? In that case the error would get overwritten.

Reply:
Yes, to prevent lose errors, maybe we could use the callback to handle the errors.

Reply:
callback added, all errors could be handled at the endpoint.